src/D_functions_and_procedures.sql

```diff
@@ -368,39 +368,55 @@ SELECT title FROM release WHERE media_id=47460;
 
 -------------------D9 SIMILAR MOVIES SEARCH-------------------------
 --------------------------------------------------------------------
-
 --FINAL FUNCTION SIMILAR MOVIES
 --THIS COMBINES OTHER FUNCTIONS WHICH ARE DECLARED LOWER IN THIS FILE
 CREATE OR REPLACE FUNCTION get_similar_movies(input_media_id INTEGER)
 RETURNS TABLE (
 	media_id INTEGER,
-	count INTEGER
+	count INTEGER,
+	title TEXT
 ) AS $$
 
 DECLARE
 	input_media RECORD;
 BEGIN
 
   --get title id and country of input_media_id
-	SELECT DISTINCT m.media_id, r.title, mpc.country_id INTO input_media
+	SELECT DISTINCT m.media_id, t.name, mpc.country_id INTO input_media
 	FROM media m
-	JOIN release r USING (media_id)
+	JOIN title t USING (media_id)
 	LEFT JOIN media_production_country mpc USING(media_id)
 	WHERE m.media_id = input_media_id;
 	
-	RETURN query
+	CREATE TEMPORARY TABLE search_result (
```

Comment:
You dont have to change it, but I think it would have been better to use the with statement here, I think that temporary tables only really should be used if you perform some kind of very expensive query. 

Also I think that the way its implemented here can create some problems if for example call the search function multiple times, then you would end up with a lot of temporary tables in the database which is not really scalable

Reply:
are not they removed immediately after the function is done? got to read into it better.